internal/httpcore/middleware.go

```diff
@@ -1,20 +1,23 @@
 package httpcore
 
 import (
-	"errors"
-	"fmt"
-
-	"github.com/CSSUoB/society-voting/internal/database"
 	"github.com/gofiber/fiber/v2"
 )
 
 func (middleware) requireAuthenticated(ctx *fiber.Ctx) error {
-	userID, authStatus := getSessionAuth(ctx)
+	authToken := ctx.Cookies(sessionTokenCookieName)
+	user, authStatus := getAuthStatus(authToken)
+
 	if authStatus == authNotAuthed {
 		return fiber.ErrUnauthorized
+	} else if authStatus == authInvalid {
```

Comment:
Since `authInvalid` could be combined with other flags, it might be worth changing this check to `authStatus & authInvalid != 0` to idiot-proof this for the future. But also this works fine as it is right now so eh, feel free to ignore.